tsweb/tsweb_test.go

```diff
@@ -650,6 +649,15 @@ func TestCleanRedirectURL(t *testing.T) {
 		{"http://myserver:31544/a/foobar", myServer, "http://myserver:31544/a/foobar", false},
 		{"http://evil.com/foo", tailscaleHost, "", true},
 		{"//evil.com", tailscaleHost, "", true},
+		{"\\\\evil.com", tailscaleHost, "", true},
+		{"javascript:alert(123)", tailscaleHost, "", true},
+		{"file:///", tailscaleHost, "", true},
+		{"file:////SERVER/directory/goats.txt", tailscaleHost, "", true},
+		{"https://google.com", tailscaleHost, "", true},
+		{"", tailscaleHost, "", true},
```

Comment:
The test case is wrong IMO. The `allowedHosts` parameter (`tailscaleHost` in this case) is a list of all hostnames you can put in the redirect URL. But a redirect URL can be a relative path, like `/admin/machines` and without a hostname.

This specific case (empty string) means redirect to `/`, which is valid. We want to support it, in particular here: https://github.com/tailscale/corp/blob/5af8524435526561c51f15faed439c9902c9bede/control/auth/signin.go#L36 (if you go through the login flow without setting `next_url` at all).

At the very bottom, there's already a test case that handles empty URL.
```suggestion
```

Reply:
How about, to document and enforce the behavior you describe,

```
{"", tailscaleHost, "", false},
```

? (It passes locally.)

Reply:
Sure!